Block/Menu.php

```diff
@@ -368,7 +368,8 @@ private function getMenuNodeBlock($node)
             ->setNodeClasses($node->getClasses())
             ->setMenuClass($this->getMenu()->getCssClass())
             ->setMenuCode($this->getData('menu'))
-            ->setTarget($node->getTarget());
+            ->setTarget($node->getTarget())
+            ->setExtraData($node->getExtraData());
```

Comment:
Could you please rename this to `setAdditionalData()` so it matches some Magento 2 naming like in eg. here: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Quote/Model/Quote/Payment.php#L341?

Also, could you please add the method to the node interface/model: https://github.com/SnowdogApps/magento2-menu/blob/develop/Api/Data/NodeInterface.php so it follows the service contract pattern?

Thanks in advance